rl4co/models/nn/env_embeddings/edge.py

```diff
@@ -0,0 +1,145 @@
+import torch
+import torch.nn as nn
+
+from torch import Tensor
+
+try:
+    from torch_geometric.data import Batch, Data
```

Comment:
I think there are two ways. Honestly, I think the current version with the import error is on the right direction! But we can make some improvements.

The first way would be to enforce `torch_geometric` installation. But this is not optimal for some users because of pretty strict requirements.
The second way (and IMO preferable) is not to enforce installation (as done now) but raise error _only_ when we try to initialize the module. We could instead import `Batch` and `Data` as `None`. Then, if the module is initialized, raise the `ImportError`.
We can follow this example: [1) import with None ](https://github.com/Dao-AILab/flash-attention/blob/6c9e60de566800538fedad2ad5e6b7b55ca7f0c5/flash_attn/modules/mlp.py#L16)and [2) raise error upon initialization](https://github.com/Dao-AILab/flash-attention/blob/6c9e60de566800538fedad2ad5e6b7b55ca7f0c5/flash_attn/modules/mlp.py#L70) )


---

PS: let's also make a `all` requirements for installing everything as [here](https://stackoverflow.com/questions/75490231/pip-pyproject-toml-can-optional-dependency-groups-require-other-optional-depend).

Reply:
> The second way (and IMO preferable) is not to enforce installation (as done now) but raise error _only_ when we try to initialize the module. We could instead import `Batch` and `Data` as `None`. Then, if the module is initialized, raise the `ImportError`. 

The second way seems more appropriate. I'll try with that one.

> PS: let's also make a `all` requirements for installing everything as [here](https://stackoverflow.com/questions/75490231/pip-pyproject-toml-can-optional-dependency-groups-require-other-optional-depend).

Does `all` need to include development dependencies like `testing` and `dev`?

Moreover, regarding optional dependencies, should we rename `graph` to `nar` to indicate that it is a dependency for the optional feature `nar`?

Reply:
I think `all` doesn't have to include `dev` and `testing` since these two parts of dependencies are only used for developers, and only called from RL4CO repository source code. 
```
testing = ["pytest", "pytest-cov"]
dev = ["black", "ruff", "pre-commit>=3.3.3"]
```
As for the `graph`, we also have the [GCN encoder](https://github.com/ai4co/rl4co/blob/main/rl4co/models/nn/graph/gcn.py) requires `torch_geometric`. Maybe we could keep it name as `graph` in the current state and later if we have good ideas to make it clearer we could update then.